agents/xy_stage/xy_latrt_agent.py

```diff
@@ -0,0 +1,253 @@
+        """
+        if params is None:
+            params = {}
+
+        
+        f_sample = params.get('sampling_frequency', self.sampling_frequency)
+        pm = Pacemaker(f_sample, quantize=True)
+
+        if not self.initialized or self.xy_stage is None:
+            raise Exception("Connection to XY Stages not initialized")
+        
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            self.log.info(f"Starting Data Acquisition for XY Stages at {f_sample} Hz")
+            session.set_status('running')
+            self.take_data = True
+            last_release = time.time()
+
+            while self.take_data:
+                if time.time()-last_release > 1.:
+                    if not self.lock.release_and_acquire(timeout=10):
+                        self.log.warn(f"Could not re-acquire lock now held by {self.lock.job}.")
+                        return False, "could not re-acquire lock"
+                    last_release = time.time()
+                pm.sleep()
+
+                data = {'timestamp':time.time(), 'block_name':'positions','data':{}}
+                pos = self.xy_stage.position
+                self.is_moving = self.xy_stage.moving
+
+                data['data']['x'] = pos[0]
+                data['data']['y'] = pos[1] 
+
+                self.agent.publish_to_feed('positions',data)
+
+        return True, 'Acquisition exited cleanly.'
+    
+    def stop_acq(self, session, params=None):
+        """
+        params: 
+            dict: {}
+        """
+        if self.take_data:
+            self.take_data = False
+            return True, 'requested to stop taking data.'
+        else:
+            return False, 'acq is not currently running.'
+    
+def make_parser(parser=None):
+    """Build the argument parser for the Agent. Allows sphinx to automatically
+    build documentation based on this function.
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    # Add options specific to this agent.
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--ip-address')
+    pgroup.add_argument('--port')
+    pgroup.add_argument('--mode')
+    pgroup.add_argument('--sampling_frequency')
+    return parser
+
+
+if __name__ == '__main__':
+    # For logging
+    txaio.use_twisted()
+    LOG = txaio.make_logger()
+
+    # Start logging
+    txaio.start_logging(level=os.environ.get("LOGLEVEL", "info"))
+
+    # Get the default ocs argument parser.
+    site_parser = site_config.add_arguments()
+
+    parser = make_parser(site_parser)
+
+    # Parse comand line.
+    args = parser.parse_args()
+    
+    # Interpret options in the context of site_config.
+    site_config.reparse_args(args, 'XY_StageAgent')
```

Comment:
`reparse_args` is deprecated. This block should instead read:
```
    parser = make_parser()

    # Interpret options in the context of site_config.
    site_config.parse_args(agent_class='XY_StageAgent', parser=parser)
```

Reply:
Do you have an example of where this is used? Reparse_args is still in the docs: https://ocs.readthedocs.io/en/latest/developer/agents.html 

Just replacing that block as-is means I don't have the args variable for the rest of that code.

Reply:
Yeah, better docs are here: https://ocs.readthedocs.io/en/develop/developer/site_config.html#agent-site-related-command-line-parameters

(Working on getting that other page updated: https://github.com/simonsobs/ocs/pull/199)

> Just replacing that block as-is means I don't have the args variable for the rest of that code.

Sorry, I missed an `args = ` before `site_config.parse_args()`.

Reply:
Found on in the smurf recorder